### ISSUE TITLE
VBA_How to end a For cycle in halfway and carry out the next one

### ISSUE THREAD

**Issue by hiddenllyy** [hiddenllyy/Internship-with-VBA#18]
Good morning,

I'd like to go through the original report table and search the key information and put it into the correct cell in my report table. Here is my basic idea:

``` vba
for each line i in my table

for each line j in the report table
    if it contains the main title and the correct zone
        for each cell k in this line j
            if there is my answer(like coulour:red)
                put it in my table
                [HERE I WANT TO QUIT THE FOR CYCLE OF k AND j AND GOTO NEXT i]
            end if
        next k
    end if
next j

next i
```

I've tried some methods but failed. So do you have some recommendations?

**Comment by mincong-h** [comment]
So what is the problem that blocks you ?

**Comment by hiddenllyy** [comment]
In the cycle `for each line j` and `for each celle k in this line`, even if I've found the information that I need, I still have to reach the last one in the cycle. And that takes time to do so, so I want to quit the `For` as soon as I've found it. And continue the `for each line i` for the next line in my table.

**Comment by hiddenllyy** [comment]
I've found it. I can exit them seperately.
`for each celle k in this line`, I can just add a `exit for` and I quit the cycle of k.

`for each line j`, I can set a flag `Dim Flag as Boolean`, and if I've found the solution, I can set the flag to true`Flag = True`, and before `next j`, put `If Flag Then Exit For`. And just don't forget to add a `Flag = Flase` to reset the flag, ortherwise it can just search the line for one time.
